### ISSUE TITLE
Incorrect type on uploadMedia for `onError`

### ISSUE THREAD

**Issue by migueliriano** [lipemat/types-js-boilerplate#2]
Using the `onError` of `uploadMedia` I noticed that is using the generic `Error` type , but uploadMedia is returning a different error structure and we have a typescript error if we try to access a different key:

Example Error structure:

```js
{
code: "SIZE_ABOVE_LIMIT",
file: File {name: 'Airbus_Pleiades_50cm_8bit_RGB_Yogyakarta.jpg', lastModified: 1643917147077, lastModifiedDate: Thu Feb 03 2022 15:39:07 GMT-0400 (Bolivia Time), webkitRelativePath: '', size: 41026764, …},
message: [
    {$$typeof: Symbol(react.element), type: 'strong', key: 'filename', ref: null, props: {…}, …}.
     ": ".
    "This file exceeds the maximum upload size for this site.",
]
}
```

*Potential solution*
Use a custom error type following the next structure:

```ts
type UploadMediaError = {
	code: string;
	file: File;
	message: ( string | object )[]
};
```
*Issue Source*: https://github.com/lipemat/types-js-boilerplate/blob/009c3877bd60f25af1220e9dca6266a7868669f4/wordpress__media-utils/index.d.ts#L32

**Comment by lipemat** [comment]
Where is the source code for what is returned? I'd like to see if it's a common type.

**Comment by migueliriano** [comment]
@lipemat - This is the source code of the return error: https://github.com/WordPress/gutenberg/blob/921e3ec436b1de48c80bec6ea9ebadfb808fe3e6/packages/media-utils/src/utils/upload-media.js#L107-L116

I see they are using the same structure each time that they call the function `triggerError`.

**Comment by lipemat** [comment]
This has been addressed in #ea7b173
